app/api/league/route.ts

```diff
@@ -18,30 +24,22 @@ const LeagueConfig = z.object({
 export async function POST(request: NextRequest) {
     // check auth
     const body = await request.json();
-    if (!body.token) {
-        return NextResponse.json({"error": unauthenticatedErrorMessage}, {status: 401})
-    }
-    const client = new OAuth2Client();
-    const clientId = process.env.GOOGLE_LOGIN_CLIENT_ID;
-    let authResponse = null;
-    try {
-        authResponse = await client.verifyIdToken({
-            idToken: body.token,
-            audience: clientId
-        });
-    }
-    catch(error) {
-        console.log(error);
-        return NextResponse.json({"error": unauthenticatedErrorMessage}, {status: 401})
-    }
-
-    const payload:TokenPayload | undefined = authResponse.getPayload();
-
-    if(payload){
-        const googleUserId = payload["sub"];
-        if (googleUserId !== "108251633753098119380") {
-            return NextResponse.json({"error": "you are not authorized to perform that action"}, {status: 403});
+    const sessionCookie = request.cookies.get("session");
+    if(sessionCookie){
+        const decryptedSessionCookie = await decrypt(sessionCookie?.value) as decryptionPayload;
+        const googleUserId = decryptedSessionCookie?.sub;
+        const invalidGoogleUserId = !googleUserId
+        const invalidGoogleUserSub = Object.keys(decryptedSessionCookie).length !== 3;
```

Comment:
TBH, I am not sure if this is the right place to verify/validate this... but I don't think that's a blocker.

Reply:
Yeah I wanted to talk to you about this as a follow up refactor